applications/addons/models/class.confidencemodel.php

```diff
@@ -0,0 +1,134 @@
+
+    private $coreVersion = null;
+    
+    public function __construct($Name = '') {
+        parent::__construct('Confidence');
+    }
+    
+    /**
+     * Get a list of addon versions of Vanilla
+     * 
+     * @param int $Limit
+     * @return array
+     */
+    public function getCoreVersions($Limit = 5) {
+        return $this->SQL->Select('a.AddonID, a.Name, av.AddonVersionID, av.Version')
+                ->From('Addon a')
+                ->Where('a.AddonTypeID', ADDON_TYPE_CORE)
+                ->Where('a.AddonKey', 'vanilla')
+                ->Where('av.Deleted', 0)
+                ->Join('AddonVersion av', 'a.AddonID = av.AddonID')
+                ->OrderBy('av.DateInserted', 'desc')
+                ->Limit($Limit)
+                ->Get()
+                ->Result();
+    }
+
+    /**
+     * Get the latest addon version of Vanilla
+     * 
+     * @return bool|stdClass False when empty, object otherwise
+     */
+    public function getCoreVersion() {
+        if (is_null($this->coreVersion)) {
+            $this->coreVersion = $this->SQL->Select('a.Name, av.AddonVersionID, av.Version')
+                ->From('Addon a')
+                ->Where('a.AddonTypeID', ADDON_TYPE_CORE)
+                ->Where('a.AddonKey', 'vanilla')
+                ->Where('av.Deleted', 0)
+                ->Join('AddonVersion av', 'a.AddonID = av.AddonID')
+                ->OrderBy('av.DateInserted', 'desc')
+                ->Limit(1)
+                ->Get()
+                ->FirstRow();
+        }
+        
+        return $this->coreVersion;
+    }
+    
+    /**
+     * Validates an ID and returns the addon version associated with the core
+     * version. Defaults to the latest core version if the ID is invalid
+     * 
+     * @param int $ID
+     * @return stdClass
+     */
+    public function checkCoreVersion($ID) {
+        $result = $this->SQL->Select('a.Name, av.AddonVersionID, av.Version')
+                ->From('Addon a')
+                ->Where('a.AddonTypeID', ADDON_TYPE_CORE)
+                ->Where('a.AddonKey', 'vanilla')
+                ->Where('av.AddonVersionID', $ID)
+                ->Join('AddonVersion av', 'a.AddonID = av.AddonID')
+                ->OrderBy('av.DateInserted', 'desc')
+                ->Limit(1)
+                ->Get()
+                ->FirstRow();
+        
+        if (!$result) {
+            $result = $this->getCoreVersion();
+        }
+        
+        return $result;
+    }
+    
+    /**
+     * Loads the confidence summary of the specified addon version against the 
+     * specified core version.
+     * 
+     * @param int $AddonVersionID
+     * @param string $DataType DATASET_TYPE_ARRAY or DATASET_TYPE_OBJECT
+     * @return array|stdClass matching $DataType
+     */
+    public function getID($AddonVersionID, $coreVersionID = false, $DataType = DATASET_TYPE_ARRAY) {
+        $coreVersion = $this->checkCoreVersion($coreVersionID);
+        $Confidence = $this->SQL
```

Comment:
I'd be more OK with using the old coding standards if you weren't randomly switching between casings. :)